pkg/solana/logpoller/loader_test.go

```diff
@@ -0,0 +1,289 @@
+			blockProducer.makeEvent()
+		}
+	}
+
+	b.ReportMetric(float64(parser.Count())/b.Elapsed().Seconds(), "events/sec")
+	b.ReportMetric(float64(blockProducer.Count())/b.Elapsed().Seconds(), "rcp_calls/sec")
+}
+
+type testBlockProducer struct {
+	b *testing.B
+
+	mu        sync.RWMutex
+	nextSlot  uint64
+	blockSigs map[uint64][]solana.Signature
+	sigs      map[string]bool
+	count     uint64
+}
+
+func (p *testBlockProducer) incrementSlot() {
+	p.b.Helper()
+
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	p.nextSlot++
+	p.blockSigs[p.nextSlot] = make([]solana.Signature, 0, 100)
+}
+
+func (p *testBlockProducer) makeEvent() {
+	p.b.Helper()
+
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	sig := solana.Signature{byte(rand.Int64())}
+
+	p.blockSigs[p.nextSlot] = append(p.blockSigs[p.nextSlot], sig)
+	p.sigs[sig.String()] = true
+}
+
+func (p *testBlockProducer) Count() uint64 {
+	p.mu.RLock()
+	defer p.mu.RUnlock()
+
+	return p.count
+}
+
+func (p *testBlockProducer) GetLatestBlockhash(_ context.Context, _ rpc.CommitmentType) (out *rpc.GetLatestBlockhashResult, err error) {
+	p.b.Helper()
+
+	p.mu.Lock()
+	p.count++
+	p.mu.Unlock()
+
+	p.mu.RLock()
+	defer p.mu.RUnlock()
+
+	return &rpc.GetLatestBlockhashResult{
+		RPCContext: rpc.RPCContext{
+			Context: rpc.Context{
+				Slot: p.nextSlot,
+			},
+		},
+	}, nil
+}
+
+func (p *testBlockProducer) GetBlocks(_ context.Context, startSlot uint64, endSlot *uint64, _ rpc.CommitmentType) (out rpc.BlocksResult, err error) {
+	p.b.Helper()
+
+	p.mu.Lock()
+	p.count++
+	p.mu.Unlock()
+
+	blocks := make([]uint64, *endSlot-startSlot)
+	for idx := range blocks {
+		blocks[idx] = startSlot + uint64(idx)
+	}
+
+	return rpc.BlocksResult(blocks), nil
+}
+
+func (p *testBlockProducer) GetBlockWithOpts(_ context.Context, block uint64, opts *rpc.GetBlockOpts) (*rpc.GetBlockResult, error) {
+	p.b.Helper()
+
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	var result rpc.GetBlockResult
+
+	sigs := p.blockSigs[block]
+
+	switch opts.TransactionDetails {
+	case rpc.TransactionDetailsFull:
+		result.Transactions = make([]rpc.TransactionWithMeta, len(sigs))
+		for idx, sig := range sigs {
+			delete(p.sigs, sig.String())
+
+			result.Transactions[idx] = rpc.TransactionWithMeta{
+				Slot: block,
+				Meta: &rpc.TransactionMeta{
+					LogMessages: messages,
+				},
+			}
+		}
+	case rpc.TransactionDetailsSignatures:
+		result.Signatures = sigs
+		delete(p.blockSigs, block)
+	case rpc.TransactionDetailsNone:
+		fallthrough
+	default:
+	}
+
+	p.count++
+	result.BlockHeight = &block
+
+	return &result, nil
+}
+
+func (p *testBlockProducer) GetSignaturesForAddressWithOpts(context.Context, solana.PublicKey, *rpc.GetSignaturesForAddressOpts) ([]*rpc.TransactionSignature, error) {
+	p.b.Helper()
+
+	return nil, nil
+}
+
+func (p *testBlockProducer) GetTransaction(_ context.Context, sig solana.Signature, _ *rpc.GetTransactionOpts) (*rpc.GetTransactionResult, error) {
+	p.b.Helper()
+
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	var msgs []string
+
+	p.count++
```

Comment:
I'm not sure I understand what `p.count` is supposed to represent.  Looks like it is incremented when some methods of `testBlockProducer` are called (eg. `GetTransaction`, `GetBlocks`) but not when other methods are called (eg. `GetSignatureForAddressWithOpts`)

Reply:
I guess `GetSignaturesForAddressWithOpts` is the only one that doesn't call it, so maybe just unimplemented. Still, does it make sense for us to be adding together counts from many different methods? Seems like it would be more useful to track them each individually.

Reply:
`GetSignatureForAddressWithOpts` is for backfilling and that currently isn't part of the benchmark test. `p.count` increments the simulated rpc call count and is used to measure the call rate in the benchmark test.